homeassistant/components/unifiled/config_flow.py

```diff
@@ -0,0 +1,64 @@
+"""Config flow for unifiled integration."""
+import logging
+
+from unifiled import unifiled
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({"host": str, "port": str, "username": str, "password": str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    try:
+        unifiled(
+            data["host"],
+            data["port"],
+            username=data["username"],
+            password=data["password"],
+        )
+        return {"title": data["host"]}
+    except Exception as ex:  # pylint: disable=broad-except
+        _LOGGER.error(ex)
+        raise CannotConnect()
+
+
+class DomainConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for unifiled."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+
+                return self.async_create_entry(title=info["title"], data=user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
```

Comment:
As above, this code won't be reached currently it seems.

Reply:
As above